src/webdriver.rs

```diff
@@ -188,3 +188,65 @@ where
         }
     }
 }
+
+#[cfg(feature = "persist")]
+pub mod persist {
+    use std::marker::PhantomData;
+
+    use serde::{Deserialize, Serialize};
+
+    use crate::{
+        common::types::SessionId,
+        error::WebDriverResult,
+        http::connection_async::WebDriverHttpClientAsync,
+        session::{spawn_session_task, WebDriverSession},
+        webdriver::{GenericWebDriver, WebDriver},
+    };
+
+    /// The persisted session that can be serialized and deserialized.
+    ///
+    /// # TODO
+    ///
+    ///  - What is `WebDriverConfig` used for? Do we need to persist it as well?
+    ///
+    #[derive(Clone, Debug, Serialize, Deserialize)]
+    pub struct PersistedSession {
+        session_id: SessionId,
+        capabilities: serde_json::Value,
+    }
+
+    impl PersistedSession {
+        pub async fn connect<T>(
+            self,
+            remote_server_addr: &str,
+        ) -> WebDriverResult<GenericWebDriver<T>>
+        where
+            T: WebDriverHttpClientAsync + 'static,
+        {
+            let conn = T::create(remote_server_addr)?;
+            // TODO: Check if session exists?
```

Comment:
It's probably worthwhile to check that the session still exists at this point. Perhaps attempt get the current URL and if that fails you could bubble the error up to the caller. That then gives the caller the assurance that if this succeeds, you're good to go.

Reply:
Yes, getting the current URL might be a good test.